os/board/rtl8720e/src/component/os/tizenrt/rtk_blemgr.c

```diff
@@ -0,0 +1,439 @@
+
+/*** Advertiser(Broadcaster) ***/
+trble_result_e trble_netmgr_set_adv_data(struct bledev *dev, trble_data *data);
+trble_result_e trble_netmgr_set_adv_resp(struct bledev *dev, trble_data *data);
+trble_result_e trble_netmgr_set_adv_type(struct bledev *dev, trble_adv_type_e adv_type, trble_addr *addr);
+trble_result_e trble_netmgr_set_adv_interval(struct bledev *dev, uint16_t interval);
+trble_result_e trble_netmgr_start_adv(struct bledev *dev);
+trble_result_e trble_netmgr_stop_adv(struct bledev *dev);
+
+struct trble_ops g_trble_drv_ops = {
+	// Common
+	trble_netmgr_init,
+	trble_netmgr_deinit,
+	trble_netmgr_get_mac_addr,
+	trble_netmgr_get_bonded_device,
+	trble_netmgr_delete_bond,
+	trble_netmgr_delete_bond_all,
+	trble_netmgr_conn_is_active,
+	trble_netmgr_conn_is_any_active,
+	trble_netmgr_ioctl,
+
+	// Observer
+	trble_netmgr_start_scan,
+	trble_netmgr_stop_scan,
+	trble_netmgr_scan_whitelist_add,
+	trble_netmgr_scan_whitelist_delete,
+	trble_netmgr_scan_whitelist_clear_all,
+
+	// Client
+	trble_netmgr_client_connect,
+	trble_netmgr_client_disconnect,
+	trble_netmgr_client_disconnect_all,
+	trble_netmgr_connected_device_list,
+	trble_netmgr_connected_info,
+	trble_netmgr_operation_enable_notification,
+	trble_netmgr_operation_read,
+	trble_netmgr_operation_write,
+	trble_netmgr_operation_write_no_response,
+
+	// Server
+	trble_netmgr_get_profile_count,
+	trble_netmgr_charact_notify,
+	trble_netmgr_attr_set_data,
+	trble_netmgr_attr_get_data,
+	trble_netmgr_attr_reject,
+	trble_netmgr_server_disconnect,
+	trble_netmgr_get_mac_addr_by_conn_handle,
+	trble_netmgr_get_conn_handle_by_addr,
+
+	// Broadcaster
+	trble_netmgr_set_adv_data,
+	trble_netmgr_set_adv_resp,
+	trble_netmgr_set_adv_type,
+	trble_netmgr_set_adv_interval,
+	trble_netmgr_start_adv,
+	trble_netmgr_stop_adv,
+};
+
+// temporary remove
+// extern int rltk_coex_set_ble_scan_duty(uint8_t duty);
+trble_result_e trble_netmgr_init(struct bledev *dev, trble_client_init_config *client, trble_server_init_config *server)
+{
+	printf("[%s : %d]\n", __FILE__, __LINE__);
+	trble_result_e ret = TRBLE_INVALID_ARGS;
+	// temporary remove
+	// if (!rltk_wlan_running(0)) {
+	// 	TRBLE_TEST_ERR("[TRBLE] WiFi is off. Please turn on WiFi first.\n");
+	// 	return TRBLE_FAIL;
+	// }
+#ifdef CONFIG_AMEBALITE_BLE_SCATTERNET
+	if (client != NULL && server != NULL) {
+		ret = rtw_ble_combo_init(client, server);
+	}
+#elif defined(CONFIG_AMEBALITE_BLE_CENTRAL)
+	if (client != NULL) {
+		ret = rtw_ble_client_init(client);
+	}
+#elif defined(CONFIG_AMEBALITE_BLE_PERIPHERAL)
+	if (server != NULL) {
+		ret = rtw_ble_server_init(server);
+	}
+#endif
+	/* Set WiFi time slot */
+	// temporary remove
+	// (void)rltk_coex_set_ble_scan_duty(WIFI_TIME_SLOT);
+	if (ret == TRBLE_SUCCESS) {
+		ret = rtw_ble_server_get_mac_address(dev->hwaddr);
+
+		if (ret != TRBLE_SUCCESS) {

```

Comment:
In this case, this function returns failure.
Is it intended?

Reply:
Oh, you are right.  We need to check with a new variable for mac api. 
This will be fixed other PR in both rtl8720e and rtl8721csm.

Reply:
Do you mean rtw_ble_server_get_mac_address will always return failure?